homeassistant/components/ecobee/__init__.py

```diff
@@ -73,6 +78,24 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
 
     await hass.config_entries.async_forward_entry_setups(entry, PLATFORMS)
 
+    # The legacy Ecobee notify.notify service is deprecated
+    # was with HA Core 2024.5.0 and will be removed with HA core 2024.7.0
+    entry_options: dict[str, str | int | float | None] = dict(entry.options or {})
+    if entry_options.get(CONF_MIGRATE_NOTIFY):
+        return True
+
+    ir.async_create_issue(
+        hass,
+        DOMAIN,
+        "migrate_notify",
+        breaks_in_ha_version="2024.7.0",
```

Comment:
What's the deprecation period for this?

Reply:
Would 2 months not be enough for users to update their automations?

Reply:
That would be fine too IMO